### PR TITLE
Paramaterize test_minimal_runtime_hello_world test. NFC.

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

**Comment by sbc100** [comment]
@juj expressed some concern over on #10301 about the general use of parameterised tests.   I have to say I think they are so so much better than tests that internally loop over different test cases.
  I love being able to run each individually and see each on potentially fail individually.  

I'm not saying we should use them everywhere, but certainly where it makes sense. Especially for cases like this where there is single loop in the test body. 

We do have tests with multiple nested loops which might not be so applicable, but I also find such nested loops in tests code kind of terrifying and would love to have fewer such tests going forward if possible.

In short, I'm a huge fan of being able to split up these tests.. I hope that you can agree it makes sense at least in these types of cases @juj ?

**Comment by juj** [comment]
My thoughts were already expressed in that earlier comment: I find them overly verbose to read and maintain, hard to search, and unable to expand with nested permutations.

Agree that the ability to run a single subtest is a very good point towards these.

I don't know what to do here. lgtm to land here if you'd like.

**Comment by kripken** [comment]
Personally I think parametrization is good when a test has just one big loop. When tests have multiple loops or other complex stuff then it's not helpful.
